drivers/sensors/pmw3389.c

```diff
@@ -0,0 +1,289 @@
+// clang-format off
+#define REG_Product_ID                 0x00
+#define REG_Revision_ID                0x01
+#define REG_Motion                     0x02
+#define REG_Delta_X_L                  0x03
+#define REG_Delta_X_H                  0x04
+#define REG_Delta_Y_L                  0x05
+#define REG_Delta_Y_H                  0x06
+#define REG_SQUAL                      0x07
+#define REG_RawData_Sum                0x08
+#define REG_Maximum_RawData            0x09
+#define REG_Minimum_RawData            0x0A
+#define REG_Shutter_Lower              0x0B
+#define REG_Shutter_Upper              0x0C
+#define REG_Ripple_Control             0x0D
+#define REG_Resolution_L               0x0E
+#define REG_Resolution_H               0x0F
+#define REG_Config2                    0x10
+#define REG_Angle_Tune                 0x11
+#define REG_Frame_Capture              0x12
+#define REG_SROM_Enable                0x13
+#define REG_Run_Downshift              0x14
+#define REG_Rest1_Rate_Lower           0x15
+#define REG_Rest1_Rate_Upper           0x16
+#define REG_Rest1_Downshift            0x17
+#define REG_Rest2_Rate_Lower           0x18
+#define REG_Rest2_Rate_Upper           0x19
+#define REG_Rest2_Downshift            0x1A
+#define REG_Rest3_Rate_Lower           0x1B
+#define REG_Rest3_Rate_Upper           0x1C
+#define REG_Observation                0x24
+#define REG_Data_Out_Lower             0x25
+#define REG_Data_Out_Upper             0x26
+#define REG_SROM_ID                    0x2A
+#define REG_Min_SQ_Run                 0x2B
+#define REG_RawData_Threshold          0x2C
+#define REG_Control2                   0x2D
+#define REG_Config5_L                  0x2E
+#define REG_Config5_H                  0x2F
+#define REG_Power_Up_Reset             0X3A
+#define REG_Shutdown                   0x3B
+#define REG_Inverse_Product_ID         0x3F
+#define REG_LiftCutoff_Cal3            0x41
+#define REG_Angle_Snap                 0x42
+#define REG_LiftCutoff_Cal1            0x4A
+#define REG_Motion_Burst               0x50
+#define REG_SROM_Load_Burst            0x62
+#define REG_Lift_Config                0x63
+#define REG_RawData_Burst              0x64
+#define REG_LiftCutoff_Cal2            0x65
+#define REG_LiftCutoff_Cal_Timeout     0x71
+#define REG_LiftCutoff_Cal_Min_Length  0x72
+#define REG_PWM_Period_Cnt             0x73
+#define REG_PWM_Width_Cnt              0x74
+
+// clang-format on
+
+#ifndef MAX_CPI
+#    define MAX_CPI 0x013F  // limits to 0--319, should be max cpi/50 -1
+#endif
+
+bool _inBurst = false;
+
+#ifdef CONSOLE_ENABLE
+void print_byte(uint8_t byte) { dprintf("%c%c%c%c%c%c%c%c|", (byte & 0x80 ? '1' : '0'), (byte & 0x40 ? '1' : '0'), (byte & 0x20 ? '1' : '0'), (byte & 0x10 ? '1' : '0'), (byte & 0x08 ? '1' : '0'), (byte & 0x04 ? '1' : '0'), (byte & 0x02 ? '1' : '0'), (byte & 0x01 ? '1' : '0')); }
+#endif
+#define constrain(amt, low, high) ((amt) < (low) ? (low) : ((amt) > (high) ? (high) : (amt)))
+
+bool spi_start_adv(void) {
```

Comment:
would be great to keep the names similar between adns and the pmw3360. You might not have seen PR#15559 yet?

Reply:
names updated in pmw3389.h and pmw3389.c
missed that PR should be up to date now.